CONTRIBUTING.md

```diff
@@ -61,6 +61,39 @@ This basically builds `terracognita` with the current sources.
 You also need to install other code dependencies not mandatory in the runtime environment:
   * [enumer](https://github.com/dmarkham/enumer) is used to generate some code.
   * [goimports](https://godoc.org/golang.org/x/tools/cmd/goimports) is used to format / organize code and imports. CI will perform a check on this, we highly recommend to run `$ goimports -w <your-modified-files>`.
+  * [mockgen](https://github.com/golang/mock) is used to generates the mocks for testing
+
+#### Add a new Provider
+
+:warning: It has to support `github.com/hashicorp/terraform-plugin-sdk@v2.0.0` or higher, providers with `v1` will not work anymore :warning:
+
+All the provider names of packages and so one are the ones used by the provider itself, the one in which all the resources are prefixed with (ex: `aws_instance` is `aws`)
+
+**Provider**
+
+All providers have to follow the [provider.Provider](https://github.com/cycloidio/terracognita/blob/master/provider/provider.go#L14) interface in order to be able to be integrated to the codebase. The best way would be to follow any of the other providers and modify it accordingly to your needs. Also reading the documentation of the interface will help a lot in any question regarding what means any of the functions.
+
+New providers have to be on the root with the package name being the name of the provider on Terraform (ex: Amazon is `aws`)
+
+The `NewProvider` will need all the information required to initialize the specific provider client(sdk), and the specific provider configuration so then Terraform can use it to read information remotely and load it to the Schemas. All the information has to be sent via parameters, no ENV variables.
+
+**Resources**
+
+Resources are each of one of the Terraform resources that this specific provider has and can import using TerraCognita. The main idea is that the `provider.Provider.Resources` returns all the instances of one resource. That information has to be fetched from the provider API using the sdk initialized before and for each instance we have to build the Terraform resource ID and return the resource. This then will be used by Terraform to read all the needed data from the provider to load the resource Schema (no need to do anything as this is abstracted into the caller of that function).
+
+To know what's the ID expected by Terraform you can check on the Terraform documentation of the resource ([doc](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/instance#import)[TODO: link docs and tf code], sometimes is more complex and it needs several elements to build an ID ([doc](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/iam_role_policy#import) and [code](https://github.com/hashicorp/terraform-provider-aws/blob/main/aws/resource_aws_iam_role_policy.go#L171)) so we have to set it on the right format for then be read for Terraform.
+
```

Comment:
I'd add a paragraph for the "multiple style resources" - eg. `security_group` and `security_group_rules` - which one to pick, how to decide, etc.